src/BlazorTurnstile/TurnstileAppearance.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Runtime.Serialization;
+using System.Text.Json.Serialization;
+
+namespace BlazorTurnstile;
+
+[JsonConverter(typeof(JsonStringEnumConverter))]
+public enum TurnstileAppearance {
+    always,
+    execute,
+    [EnumMember(Value = "interaction-only")]
```

Comment:
It seems like the `EnumMemeberAttribute` isn't yet supported and the sample won't work without additional converters. The interaction only value is not serialized correctly which makes turnstile js to throw an exception.  Did you get this working on your end?

Reply:
Thanks for looking into this, @MichelJansson.
I apologize for this hiccup and you are correct. The serialization does not respect enumerations without converters and went with a more classical approach:

```javascript
    // Convert non-callback parameter(s) with values containing "_" to expected values with "-" because serializations do not respect enumerations.
    for (let key in parameters) {
         if (!["call-back", "action", "cData", "response-field-name"].includes(key)) {
            if (parameters[key] && typeof parameters[key] === 'string' && parameters[key].includes("_")) {
                parameters[key] = parameters[key].replace(/_/g, "-");
            }
        }
    }
``` 

This should work for any parameters with dashes in non-custom values.

Please let me know if this works for you.

Reply:
Thanks for the suggestion, but I'd like to stay in .NET land as far as possible, so I made a hole "re-take" on the serialization that now is kebab-case by default for the entire parameter object tree. This removes the need for attributes and js.